### PR TITLE
D2K - Seperate Imperial and Harkonnen Sardaukars

### LINKED ISSUE #7332: D2k - create separate Sardaukar actor for house Corrino
I have not had a chance to check this, but it has been mentioned here on GitHub before, and I saw it in several Dune 2000 wiki sites - the Emperor sardaukar are supposed to be better than the Harkonnen ones. This will require we make such an actor for house Corrino with better stats.
How much better and in what way remains to be determined.
I am leaving this here for easier tracking of remaining tasks.
Anyone can feel free to do it if they feel like it.
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/7754690-d2k-create-separate-sardaukar-actor-for-house-corrino?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by MustaphaTR**
Fixes #7332

Also fixes that Harkonnen Sardaukar had wrong/Imperial's Build Time and ROF.

Also edited BuildPaletteOrder for Inf Tabs.

### CODE REVIEW

File: OpenRA.Mods.D2k/UtilityCommands/D2kMapImporter.cs [hunk 1]
```diff
@@ -84,7 +84,7 @@ public class D2kMapImporter
 			{ 360, Pair.New("light_inf", "Harkonnen") },
 			{ 361, Pair.New("trooper", "Harkonnen") },
 			{ 362, Pair.New("fremen", "Harkonnen") },
-			{ 363, Pair.New("sardaukar", "Harkonnen") },
+			{ 363, Pair.New("mpsardaukar", "Harkonnen") },
```
Comment (reaperrr) [inline OpenRA.Mods.D2k/UtilityCommands/D2kMapImporter.cs]: Are you sure this shouldn't stay "sardaukar"? This is mostly for singleplayer maps afterall, so I'd expect most of the initital sardaukar in missions to be of the imperial variety.

**Review by reaperrr — DISMISSED** [review]

**Comment by reaperrr** [comment]
Looks good to me once my comment has been adressed (either through reasoning or PR update).

File: OpenRA.Mods.D2k/UtilityCommands/D2kMapImporter.cs (hunk 1 above)
Comment (MustaphaTR) [reply OpenRA.Mods.D2k/UtilityCommands/D2kMapImporter.cs]: Some units share entries in original map format and Harkonnen gets MP Sardaukar there, i'm sure. You can't give Harkonnen a preplaced Imperial Sardaukar in original, at least not with D2K+ Map Editor, not sure if there is even hackier way.
resim

File: mods/d2k/rules/infantry.yaml [hunk 2]
```diff
@@ -200,13 +200,13 @@ sardaukar:
 	Inherits@AUTOTARGET: ^AutoTargetGroundAssaultMove
 	Buildable:
 		Queue: Infantry
-		BuildPaletteOrder: 80
-		Prerequisites: ~barracks.harkonnen, upgrade.barracks, high_tech_factory, ~techlevel.medium
+		BuildPaletteOrder: 50
+		Prerequisites: ~palace.sardaukar, ~techlevel.high
 		BuildDuration: 81
 		BuildDurationModifier: 40
 		Description: Elite assault infantry\n  Strong vs Infantry, Vehicles\n  Weak vs Artillery
```
Comment (ltem) [inline mods/d2k/rules/infantry.yaml]: Could you add something to the description to tell them apart? E.g. `Corrino elite assault ...`

File: mods/d2k/rules/infantry.yaml [hunk 3]
```diff
@@ -229,6 +229,22 @@ sardaukar:
 		EmptyWeapon: SardDeath
 		Chance: 100
 
+mpsardaukar:
```
Comment (ltem) [inline mods/d2k/rules/infantry.yaml]: It seems that the Skirmish AI isn't able to build Sardaukars as Harkonnen due to https://github.com/OpenRA/OpenRA/blob/bleed/mods/d2k/rules/ai.yaml#L54. So it needs to be updated.

**Comment by ltem** [comment]
Shellmap:
- The cost of the Sardaukars is set to 0 ([Link](https://github.com/OpenRA/OpenRA/blob/bleed/mods/d2k/maps/shellmap/rules.yaml#L63)) but Corrino won't be able to build them (no palace). So the line should be updated to mpsardaukars to be useable for the Harkonnen.
- Small nit: This [Harkonnen Sardaukar](https://github.com/OpenRA/OpenRA/blob/bleed/mods/d2k/maps/shellmap/map.yaml#L659) should be changed to `mpsardaukar`

**Comment by MustaphaTR** [comment]
Updated.

**Review by ltem — APPROVED** [review]

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(Next-Playtest)/_compare/224645b479b6f2d9a51faf2fb369875e0f4d9d81...fcac9c81321879f2ea928e6f3131dd4fa1fb326d)

https://github.com/OpenRA/OpenRA/commit/904b71b890ba466e56bcf800c7cdb3954da53fb3

**Pull request merged**